src/distributions/utils.rs

```diff
@@ -162,3 +162,195 @@ macro_rules! simd_less_then {
 #[cfg(feature="simd_support")] simd_less_then! { f64x2 }
 #[cfg(feature="simd_support")] simd_less_then! { f64x4 }
 #[cfg(feature="simd_support")] simd_less_then! { f64x8 }
+
+
+// Until portable shuffles land in stdsimd, we expose and use the shuffle intrinsics directly.
+#[cfg(feature = "simd_support")]
+extern "platform-intrinsic" {
+    fn simd_shuffle2<T, U>(a: T, b: T, indices: [u32; 2]) -> U;
+    fn simd_shuffle4<T, U>(a: T, b: T, indices: [u32; 4]) -> U;
+    fn simd_shuffle8<T, U>(a: T, b: T, indices: [u32; 8]) -> U;
+    fn simd_shuffle16<T, U>(a: T, b: T, indices: [u32; 16]) -> U;
+    fn simd_shuffle32<T, U>(a: T, b: T, indices: [u32; 32]) -> U;
+    fn simd_shuffle64<T, U>(a: T, b: T, indices: [u32; 64]) -> U;
+}
+
+/// Implement byte swapping for SIMD vectors
+#[cfg(feature = "simd_support")]
+trait SwapBytes {
+    /// `swap_bytes` for a vector (horizontally)
+    fn swap_bytes(self) -> Self;
+}
+
+// `simd_shuffleX` require constant indices, making this a small pain to implement
+#[cfg(feature = "simd_support")]
+macro_rules! impl_swap_bytes {
+    ($ty:ident, $vec8:ident, $shuf:ident, $indices:expr) => (
+        impl SwapBytes for $ty {
+            fn swap_bytes(self) -> Self {
+                let vec8 = $vec8::from_bits(self);
+                let shuffled: $vec8 = unsafe { $shuf(vec8, vec8, $indices) };
+                $ty::from_bits(shuffled)
+            }
+        }
+    );
+
+    // bulk impl for a shuffle intrinsic/vector width
+    ($vec8:ident, $shuf:ident, $indices:expr, $($ty:ident,)+) => ($(
+        impl_swap_bytes! { $ty, $vec8, $shuf, $indices }
```

Comment:
Changing the argument order here is confusing. Keep the type at the end?

Reply:
It makes the macro invitation prettier but yeah I see how it could trip people up